src/DurableTask.Netherite/StorageLayer/Faster/AzureBlobs/AzureStorageDevice.cs

```diff
@@ -500,11 +534,17 @@ async Task ReadFromBlobAsync(UnmanagedMemoryStream stream, BlobUtilsV12.PageBlob
         {
             using (stream)
             {
+                // we use this to prevent reading past the end of the page blob
```

Comment:
I don't immediately understand why enlarging the blobs can suddenly cause reads to go past the end of the page blob. Can you please call out the connection?

Reply:
Because these blobs don't actually have a fixed size from the perspective of FASTER, the FASTER reads may easily read past the end of the blob. This is not new - we just never saw it because 512GB was always way larger than needed.

Reply:
I see. I think it would be good to write that in the inlined comment being referenced here. This also answers my question here: https://github.com/microsoft/durabletask-netherite/pull/329#discussion_r1378209082